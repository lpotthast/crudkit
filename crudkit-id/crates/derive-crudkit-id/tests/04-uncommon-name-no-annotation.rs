#![allow(dead_code)]

use derive_crudkit_id::CkId;

#[derive(CkId)]
pub struct Foo {
    foo: i32,
}

fn main() {}
