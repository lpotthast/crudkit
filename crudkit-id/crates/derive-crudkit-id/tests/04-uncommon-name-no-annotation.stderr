error: To derive CkId, at least one id field must exist.

         = help: A field is an id field if it is (a) named "id" or (b) annotated with `#[ck_id(id)]`, both marking the field as part of the entities id. Specify id fields or remove the derive, if no id fields can be defined for this entity.

 --> tests/04-uncommon-name-no-annotation.rs:5:10
  |
5 | #[derive(CkId)]
  |          ^^^^
  |
  = note: this error originates in the derive macro `CkId` (in Nightly builds, run with -Z macro-backtrace for more info)
